### PR TITLE
docs: improved supported workflow systems

### PR DESCRIPTION
**Pull request by audrium**
closes https://github.com/reanahub/docs.reana.io/issues/101

### CODE REVIEW

File: docs/running-workflows/supported-systems/serial/index.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
 # Serial
 
-A simple serial or sequential workflow system.
+The Serial workflow system is using the sequence workflow pattern where each
+computational step can run in a different containerised environment and the
+outputs of previous steps are passed along as inputs to later step. The next
```
Comment (mvidalgarcia) [inline docs/running-workflows/supported-systems/serial/index.md:5]: > outputs of previous steps are passed along as inputs to later step.

Perhaps we should use "later step**s**", in the same way we use the plural to relate to "previous steps", WDYT?

File: docs/running-workflows/supported-systems/serial/index.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
...
+step is always dependent on the previous step having already been completed.
```
Comment (mvidalgarcia) [inline docs/running-workflows/supported-systems/serial/index.md:6]:
```suggestion
step always depends on the previous step having already been completed.
```

Maybe a bit clearer? Not 100% sure though.

File: docs/running-workflows/supported-systems/serial/index.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
 # Serial
 
-A simple serial or sequential workflow system.
+The Serial workflow system is using the sequence workflow pattern where each
```
Comment (mvidalgarcia) [inline docs/running-workflows/supported-systems/serial/index.md:3]: To better illustrate Serial workflows, what do you think about including a simple DAG, mentioning inputs/outputs?

Something along this line:

```mermaid
stateDiagram-v2
  [*] --> step_1: inputs
  step_1 --> step_2: step_1 outputs
  step_2 --> step_3: step_2 outputs
  step_3 --> [*]: final result
```

Let me know, peharps is overkill.

File: docs/running-workflows/supported-systems/yadage/index.md [hunk 4]
```diff
@@ -1,3 +1,23 @@
 # Yadage
 
-The Yadage workflow system originated in particle physics community.
+The Yadage workflow system has been developed in the context of High Energy
+Physics analyses in order to realized re-usable analyses for systematic
```
Comment (mvidalgarcia) [inline docs/running-workflows/supported-systems/yadage/index.md:4]:
```suggestion
Physics analyses in order to realise re-usable analyses for systematic
```

or perhaps "perform".

I'm sure @tiborsimko has a better feedback for these descriptions.

File: docs/running-workflows/supported-systems/snakemake/index.md [hunk 5]
```diff
@@ -7,4 +7,13 @@ language, and allows them to be scaled without the need for modifications.
 Snakemake workflows can describe the software required to run them, which is automatically
 deployed in any execution environment.
 
-For additional information, see [Snakemake Documentation](https://snakemake.readthedocs.io).
+For additional information, please see:
+
+- Snakemake documentation: [https://snakemake.readthedocs.io](https://snakemake.readthedocs.io)
```
Comment (mvidalgarcia) [inline docs/running-workflows/supported-systems/snakemake/index.md:12]: Should we mention the "whether REANA offers any cool novelties such as Snakemake-on-HTC-and-HPC." [bit](https://github.com/reanahub/docs.reana.io/issues/101#issue-1101780844) as part of this task? Or does it belong to a different PR?

File: docs/running-workflows/supported-systems/yadage/index.md [hunk 6]
```diff
@@ -1,3 +1,23 @@
...
+reinterpretation of data collected by the Large Hadron Collider. The engine
+orchestrates the execution of a directed acyclic graph of containerized
```
Comment (mvidalgarcia) [inline docs/running-workflows/supported-systems/yadage/index.md:6]:
```suggestion
orchestrates the execution of a directed acyclic graph of containerised
```

File: docs/running-workflows/supported-systems/yadage/index.md [hunk 7]
```diff
@@ -1,3 +1,23 @@
...
+workloads.
+
+For additional information, please see:
+
+- Yadage documentation: [https://yadage.readthedocs.io](https://yadage.readthedocs.io)
+- Yadage tutorial: [https://yadage.github.io/tutorial](https://yadage.github.io/tutorial/)
+
+## Supported versions
+
+| REANA version         | Yadage    | Packtivity
```
Comment (mvidalgarcia) [inline docs/running-workflows/supported-systems/yadage/index.md:16]: If `Packtivity` is relevant here, perhaps Adage as well?

**Review by mvidalgarcia — COMMENTED** [review]
Would it make sense to link some specific demo specs to each workflow system? So apart from the explanation, the user could explore a simple demo spec right away. I'm not sure, perhaps it's too much.

File: docs/running-workflows/supported-systems/serial/index.md (hunk 3 above)
Comment (audrium) [reply docs/running-workflows/supported-systems/serial/index.md:3]: I'm not sure, in one hand we have some attempts to illustrate Serial workflow in the [docs](https://docs.reana.io/running-workflows/what-is-workflow/), but it does not look great IMO. So it would be nice to dedicate more effort into producing nice workflow visualizations similar to the one in your example to all workflow systems, perhaps could be done in a separate issue/PR?

File: docs/running-workflows/supported-systems/yadage/index.md (hunk 7 above)
Comment (audrium) [reply docs/running-workflows/supported-systems/yadage/index.md:16]: I'm not sure, @tiborsimko, what do you think?

File: docs/running-workflows/supported-systems/snakemake/index.md (hunk 5 above)
Comment (audrium) [reply docs/running-workflows/supported-systems/snakemake/index.md:12]: Good point, it is mentioned in the linked [blog post](https://blog.reana.io/posts/2021/support-for-running-snakemake-workflows/) in `Snakemke in the REANA ecosystem` section (it's a pity it's not possible to link a specific paragraph), but it could be explicitly mentioned here as well

Comment (mvidalgarcia) [reply docs/running-workflows/supported-systems/snakemake/index.md:19]: It's actually possible, but the hyperlink is not exposed as part of the blog UI.

https://blog.reana.io/posts/2021/support-for-running-snakemake-workflows/#snakemke-in-the-reana-ecosystem

File: docs/running-workflows/supported-systems/serial/index.md (hunk 3 above)
Comment (mvidalgarcia) [reply docs/running-workflows/supported-systems/serial/index.md:3]: OK, perhaps it's enough to keep this info in the ["What is a workflow?"](https://docs.reana.io/running-workflows/what-is-workflow/) page. I had missed that.

File: docs/running-workflows/supported-systems/yadage/index.md (hunk 7 above)
Comment (tiborsimko) [reply docs/running-workflows/supported-systems/yadage/index.md:16]: Yes, it would be good to present `(yadage, adage, packtivity)` triad.  

We could actually show also `yadage-schemas` to make it a quad, but that might perhaps be unnecessary? (Could look like an overkill: yadage/packtivity should be mostly enough)
